frame/finality-tracker/src/lib.rs

```diff
@@ -1,352 +0,0 @@
-		}
-	}
-}
-
-impl<T: Trait> Module<T> {
-	fn update_hint(hint: Option<T::BlockNumber>) {
-		if !Self::initialized() {
-			<Self as Store>::RecentHints::put(vec![T::BlockNumber::zero()]);
-			<Self as Store>::OrderedHints::put(vec![T::BlockNumber::zero()]);
-			<Self as Store>::Median::put(T::BlockNumber::zero());
-
-			<Self as Store>::Initialized::put(true);
-		}
-
-		let mut recent = Self::recent_hints();
-		let mut ordered = Self::ordered_hints();
-		let window_size = cmp::max(T::BlockNumber::one(), T::WindowSize::get());
-
-		let hint = hint.unwrap_or_else(|| recent.last()
-			.expect("always at least one recent sample; qed").clone()
-		);
-
-		// prune off the front of the list -- typically 1 except for when
-		// the sample size has just been shrunk.
-		{
-			// take into account the item we haven't pushed yet.
-			let to_prune = (recent.len() + 1).saturating_sub(window_size.saturated_into::<usize>());
-
-			for drained in recent.drain(..to_prune) {
-				let idx = ordered.binary_search(&drained)
-					.expect("recent and ordered contain the same items; qed");
-
-				ordered.remove(idx);
-			}
-		}
-
-		// find the position in the ordered list where the new item goes.
-		let ordered_idx = ordered.binary_search(&hint)
-			.unwrap_or_else(|idx| idx);
-
-		ordered.insert(ordered_idx, hint);
-		recent.push(hint);
-
-		let two = T::BlockNumber::one() + T::BlockNumber::one();
-
-		let median = {
-			let len = ordered.len();
-			assert!(len > 0, "pruning dictated by window_size which is always saturated at 1; qed");
-
-			if len % 2 == 0 {
-				let a = ordered[len / 2];
-				let b = ordered[(len / 2) - 1];
-
-				// compute average.
-				(a + b) / two
-			} else {
-				ordered[len / 2]
-			}
-		};
-
-		let our_window_size = recent.len() as u32;
-
-		<Self as Store>::RecentHints::put(recent);
-		<Self as Store>::OrderedHints::put(ordered);
-		<Self as Store>::Median::put(median);
-
-		if T::BlockNumber::from(our_window_size) == window_size {
-			let now = frame_system::Module::<T>::block_number();
-			let latency = T::ReportLatency::get();
-
-			// the delay is the latency plus half the window size.
-			let delay = latency + (window_size / two);
-			// median may be at most n - delay
-			if median + delay <= now {
-				T::OnFinalizationStalled::on_stalled(window_size - T::BlockNumber::one(), median);
-			}
-		}
-	}
-}
-
-/// Called when finalization stalled at a given number.
-#[impl_trait_for_tuples::impl_for_tuples(30)]
-pub trait OnFinalizationStalled<N> {
-	/// The parameter here is how many more blocks to wait before applying
-	/// changes triggered by finality stalling.
-	fn on_stalled(further_wait: N, median: N);
-}
-
-impl<T: Trait> ProvideInherent for Module<T> {
-	type Call = Call<T>;
-	type Error = MakeFatalError<()>;
-	const INHERENT_IDENTIFIER: InherentIdentifier = INHERENT_IDENTIFIER;
-
-	fn create_inherent(data: &InherentData) -> Option<Self::Call> {
-		if let Ok(final_num) = data.finalized_number() {
```

Comment:
It should be fine to remove the inherent provider right away since the runtime module already expects that inherent data about finalized number might not be available.